### PR TITLE
feat: add concat

### PR DESCRIPTION
**Pull request by jly36963**

### CODE REVIEW

File: src/primitives.ts [hunk 1]
```diff
@@ -18,6 +18,12 @@ function charAt<T extends string, I extends number>(
   return str.charAt(index)
 }
 
+type Concat<T extends string[]> = Join<T, ''>
+
+function concat<T extends string[]>(...strings: T): Concat<T> {
+  return strings.join('') as Concat<T>
```
Comment (gustavoguichard) [inline src/primitives.ts:24]:
```suggestion
  return join(strings, '')
```

File: src/primitives.ts [hunk 2]
```diff
@@ -20,6 +20,12 @@ function charAt<T extends string, I extends number>(
   return str.charAt(index)
 }
 
+type Concat<T extends string[]> = Join<T, ''>
+
+function concat<T extends string[]>(...strings: T): Concat<T> {
+  return join(strings) as Concat<T>
```
Comment (gustavoguichard) [inline src/primitives.ts:26]: Is it required?
```suggestion
  return join(strings)
```

Comment (jly36963) [reply src/primitives.ts:26]: nope

**Review by gustavoguichard — APPROVED** [review]
LGTM

**Pull request merged**